cli/flox/doc/flox-build.md

```diff
@@ -0,0 +1,179 @@
+---
+title: FLOX-BUILD
+section: 1
+header: "Flox User Manuals"
+...
+
+```{.include}
+./include/experimental-warning.md
+```
+> Feature flag: `build`
+q
+# NAME
+
+flox-build - build packages
+
+# SYNOPSIS
+
+```
+flox [<general-options>] build
+     [-d=<path>]
+     [-L]
+     [<package>]...
+```
+
+# DESCRIPTION
+
+Build the specified `<package`> to run or verify your software locally.
+Create build artifacts from instructions in the manifest
+of the environment in `<path>`.
+
+## Manifest defined Packages
+
+Possible values for `<package>` are all keys under the `build` attribute
+in the `manifest.toml`.
+If no `<package>` is specified, flox will attempt to build all packages
+that are defined in the environment.
+
+Packages are built by running the script defined in `build.<package>.command`
+within a `bash` subshell.
+The shell will behave as if `flox activate` was run
+immidiately prior to running the build script.
+The current working directory of the script will be set to the root
+of the `git` repository in which the environment is defined.
+Consequently, _building requires your environment to be tracked
+under `git` version control_.
+
+By default, builds are run in a sandbox.
+Within the sandbox only the files tracked by git are available.
+Untracked files, files outside of the repository and notably network access
+will be restricted to encourage "pure" and reproducible builds.
+Builds can opt-out of the sandbox by setting `build.<package>.sandbox = "off"`.
+With the sandbox disabled, building is equivalent
+to running the build script manually within a shell created by `flox activate`.
+Any build can access the _results_ of other builds
+(including non-sandboxed ones) by referring to their name via `${<package>}`.
+
+To produce build results, the build script is expected
+to copy or move artifacts to a special folder referenced by `$out`.
+Alternatively, explicit artifacts can be declared via `build.<package>.files`.
```

Comment:
question blocking: I don't think it's quite clear how this works. Where do those files have to be created?

Reply:
The files attibute is a bit unclear in general.
I also think there should eb a way to specify where the files will be located in the final package.
To add to this, i'm not even sure if the files attribute is already supported in the first place.
I'll take this out for now.